src/main/java/org/jabref/gui/maintable/MainTableNameFormatter.java

```diff
@@ -51,4 +49,8 @@ public String formatNameLatexFree(final String nameToFormat) {
                         false);
         }
     }
+
+    public boolean needsFormatting() {
+        return preferencesService.getMainTableNameFormatPreferences().getDisplayStyle() != MainTableNameFormatPreferences.DisplayStyle.AS_IS;
```

Comment:
The preferences are rather slow and this method is called for every row in the main table, so this will lead to performance problems. You can cache the `getMainTableNameFormatPreferences` as a class variable (initalized in the constructor).